### PR TITLE
MDEXP-285 Provide endpoint for quick inventory instance export

### PR DESCRIPTION
**Pull request by andrii-novytskyi**
https://issues.folio.org/browse/MDEXP-285

## Purpose
To provide endpoint for quick export from inventory.

## Approach
Provided a new endpoint,  which accepts a request consisting of:
`"criteria",
"jobProfileId",
"uuid:" [],
"recordType",
"type"`.
The type could be cql or uuid. If cql, then we will take criteria and find record UUIDs from inventory by this CQL query, if type uuid, we will take uuids from an array and search records by them.
Creating file definition and uploading file- all this stuff are included in this endpoint.
Added recordType field to exportRequest and quickExportRequest - in the future, we will support export by holdings and items, and will use this field.
#### TODOS and Open Questions
For now, the UI will not send jobProfileId if default should be used. In the future, we should make jobProfileId as required in quickExportRequest.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -405,6 +437,10 @@
       "permissionName": "data-export.export.post",
       "displayName": "Starts export process",
       "description": "Entry point to start export for the given file definition using job profile"
+    },{
+      "permissionName": "data-export.quick.export.post",
+      "displayName": "Starts export process from inventory",
```
Comment (Igor-Gorchakov) [inline descriptors/ModuleDescriptor-template.json:442]: "displayName": "Starts quick export process from inventory",

File: ramls/dataExport.raml [hunk 2]
```diff
@@ -58,6 +59,34 @@ resourceTypes:
           body:
             text/plain:
               example: "Internal server error"
+  /quick-export:
+    displayName: Export manager service
+    description: Method to start quick export for MARC records from inventory
+    post:
+      description: Starts the export process
```
Comment (Igor-Gorchakov) [inline ramls/dataExport.raml:66]: description: Starts the quick export process

File: src/main/java/org/folio/service/manager/export/strategy/ExportStrategy.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.folio.service.manager.export.strategy;
+
+import io.vertx.core.Promise;
+import org.folio.service.manager.export.ExportPayload;
+
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/manager/export/strategy/ExportStrategy.java:5]: Please add java doc

File: src/main/java/org/folio/rest/impl/DataExportImpl.java [hunk 4]
```diff
@@ -147,4 +174,39 @@ public void deleteDataExportJobExecutionsById(String id, Map<String, String> oka
...
+      .onSuccess(fileDefinition -> fileUploadService.uploadFileDependsOnTypeForQuickExport(request, fileDefinition, params)
+        .onSuccess(uploadedFileDefinition -> fileUploadService.completeUploading(uploadedFileDefinition, tenantId))
+        .onSuccess(promise::complete)
+        .onFailure(ar -> promise.fail(ar.getCause()))
+        .onFailure(ar -> promise.fail(ar.getCause())))
+      .onFailure(handler -> failToFetchObjectHelper(handler.getMessage(), asyncResultHandler));
+    return promise.future();
+  }
+
+  // for now UI will not send jobProfileId, after we will support all profiles, remove this method, and make jobProfileId as required in quickExportRequest
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/rest/impl/DataExportImpl.java:197]: PLease update comments as we agreed earier

File: descriptors/ModuleDescriptor-template.json [hunk 5]
```diff
@@ -120,6 +120,38 @@
...
+            "inventory-storage.locations.collection.get",
+            "inventory-storage.location-units.libraries.collection.get",
+            "inventory-storage.location-units.campuses.collection.get",
+            "inventory-storage.location-units.institutions.collection.get",
+            "inventory-storage.loan-types.collection.get",
+            "inventory-storage.material-types.collection.get",
+            "inventory-storage.instance-types.collection.get",
+            "inventory-storage.instance-formats.collection.get",
+            "inventory-storage.electronic-access-relationships.collection.get",
+            "configuration.entries.collection.get",
```
Comment (Igor-Gorchakov) [inline descriptors/ModuleDescriptor-template.json:149]: Put this permission `"configuration.entries.collection.get"`, after `"inventory-storage.items.collection.get"`, just to do not mix it with perms for reference data, please

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json]: updated

File: ramls/dataExport.raml (hunk 2 above)
Comment (andrii-novytskyi) [reply ramls/dataExport.raml:66]: updated

**Review by Igor-Gorchakov — APPROVED** [review]

**Review by IlliaDaliek — APPROVED** [review]

**Pull request merged**